### PR TITLE
Fallback to first available context as DefaultContext if current default context key is 'Default'

### PR DESCRIPTION
**Pull request by erich-wang**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by BethanyZhou** [comment]
Hi @erich-wang , please resolve conflicts.

File: src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs [hunk 1]
```diff
@@ -71,10 +71,10 @@ public virtual IAzureContext DefaultContext
                 }
 
                 IAzureContext result = null;
-                if (DefaultContextKey == "Default" && Contexts.Any(c => c.Key != "Default"))
+                if (DefaultContextKey == Constants.DefaultValue && Contexts.Any(c => c.Key != Constants.DefaultValue))
                 {
-                    // If the default context is "Default", but there are other contexts set, remove the "Default" context to throw the below exception
-                    TryCacheRemoveContext("Default");
+                    // If the default context is "Default", but there are other contexts set, remove the "Default" context and select first avaiable context as default
+                    TryRemoveContext(Constants.DefaultValue);
```
Comment (isra-fel) [inline src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs:77]: Debug message might be helpful here
`EnqueueDebugMessage($"Incorrect default context key {DefaultContextKey} found. Trying to remove it and falling back to the first available context.");`

**Review by isra-fel — APPROVED** [review]
LGTM

**Comment by erich-wang** [comment]
Force merge because of known IC issue.

**Pull request merged**